src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useState } from "react";
+import { Modal } from "react-bootstrap";
+
+import "./EditTableTitleModal.scss";
+
+export interface EditTableTitleModalProps {
+  isShown: boolean;
+  title: string;
+  error: string;
+  onHide: () => void;
+  onCancel: () => void;
+  onSave: (newTableTitle: string) => void;
+}
+
+export const EditTableTitleModal: React.FC<EditTableTitleModalProps> = ({
+  isShown,
+  title,
+  error,
+  onSave,
+  onHide,
+  onCancel,
+}) => {
+  const [newTableTitle, setNewTableTitle] = useState<string>(title);
+
+  const saveChanges = useCallback(() => {
+    onSave(newTableTitle);
+    onHide();
+  }, [newTableTitle, onHide, onSave]);
+
+  const changeTableTitle = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setNewTableTitle(e.target.value);
+    },
+    []
+  );
```

Comment:
Do we want to be updating this state on every keypress? Would it be better to use `react-form`/similar here?

Reply:
Forms added in: https://github.com/sparkletown/sparkle/commit/c194508b129176d4c048ebb810e9ea4c59f60865